### DISCUSSION TITLE
Is it possible to cache authenticated fetch request using server components?

### DISCUSSION THREAD

**Discussion by mradmilovic** [vercel/next.js#50643] [Help]
### Summary

I need to send same authenticated fetch request from multiple server components. I stored jwt token in a cookie and I need to fetch current user data using that token. I need to use that data in multiple server components inside component tree but want to avoid prop drilling. If I call that request multiple times from different components I see that server endpoint is called everytime, meaning that de-duping is not working.

What is the best way to solve this. This is what I've tried:
```
export const me = cache(async (token: string) => {
  const result = await fetch(`${config.apiUrl}/users/me`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
      Cookie: `token=${token}`,
    },
  });

  return result.json();
});
```

**Comment by christopher-caldwell** [comment]
I believe your request is automagically opting out of caching due to meeting the last criteria on this [list](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#opting-out-of-data-caching)

> The fetch request uses Authorization or Cookie headers and there's an uncached request above it in the component tree.

I found this post while looking for a solution to this issue. Difficult to be sure exactly what is happening though.
